gap/attr.gi

```diff
@@ -345,8 +345,13 @@ end);
 
 #
 
-InstallMethod(DigraphNrEdges, "for a digraph",
-[IsDigraph], DIGRAPH_NREDGES);
+InstallMethod(DigraphNrEdges, "for a digraph", [IsDigraph],
+function(gr)
+  if IsBound(gr!.DigraphNrEdges) then
```

Comment:
Is this check really necessary? Is it the case that setting the record component in `gr` does not set the attribute with the same name, but the other way around is true, i.e. setting/getting the attribute sets the record component?

Reply:
Yes that is the case. Setting the record component does not set the attribute, but setting the attribute sets the record component.

I can get rid of this check, and everything will happen in the C code.